pkgs/servers/web-apps/piwik/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, fetchurl, makeWrapper, php }:
+
+stdenv.mkDerivation rec {
+  name = "piwik-${version}";
+  version = "3.0.4";
+
+  src = fetchurl {
+    url = "https://builds.piwik.org/${name}.tar.gz";
+    sha512 = "2i0vydr073ynv7wcn078zxhvywdv85c648hympkzicdd746g995878py9006m96iwkmk4q664wn3f8jnfqsl1jd9f26alz1nssizbn9";
+  };
+
+  phases = [ "unpackPhase" "patchPhase" "installPhase" ];
+
+  buildInputs = [ makeWrapper ];
+
+  # regarding substitutes: looks like this is just a bug / confusion of the directories, and nobod yhas tested this.
+  # PR at https://github.com/piwik/piwik/pull/11661
+  patchPhase = ''
```

Comment:
Consider `postPatch`; at least in general its nice to allow applying arbitrary patches via override.  Specifying a custom patch phase prevents that.